localstack/services/iam/iam_starter.py

```diff
@@ -273,6 +283,7 @@ def update_group(self):
         moto_iam_backend.groups[new_group_name] = moto_iam_backend.groups.pop(group_name)
         return ""
 
+    # TODO: potentially extend @patch utility to allow "conditional" patches like below ...
```

Comment:
@thrau Any thoughts on adding a "conditional" patch functionality (specifying the patch target as a string)? Would be pretty handy! 🩹

Reply:
@whummer The intent here is to allow upstream to implement this in the future or when would this usually be false? :thinking:

Reply:
Yes, pretty much - actually, we could include a warning/info message to indicate that it is already available upstream, then we can remove the patch altogether. 👍 (assumption is obviously that these features are properly covered with tests, such that a change in implementation would still work properly)

Reply:
i can see why it would be appealing to add something like that to the `@patch` decorator, to keep using the same API. but conceptually it seems a bit odd to apply a "patch" to something only if it does not exist. that's not really applying a patch, that's creating a new function. moreover, we would have to add an optional `name: str` and `if_not_exist: bool` to the sig, where the second only makes sense if we're passing the name as a string. so the decorator would have two different control paths.

the code the way it is right now seems perfectly fine to me tbh. it's clear what's going on, and we would only be able to save a couple of lines of code. i suppose we could create a method like `setattr_ifnotexist(IamResponse, "update_group", update_group)`, but that's saving one line. not sure it's worth it.